src/main/java/org/mule/tools/devkit/sonar/utils/PomUtils.java

```diff
@@ -54,6 +64,36 @@ public static boolean isDevKitConnector(MavenProject mavenProject) {
         return parent != null && parent.getGroupId().equals(DEVKIT_GROUP_ID) && (parent.getArtifactId().equals(DEVKIT_ARTIFACT_ID) || parent.getArtifactId().equals(CERTIFIED_DEVKIT_ARTIFACT_ID));
     }
 
+    public static VersionUtils setLatestVersion() {
+        VersionUtils latestVersion = new VersionUtils("3.0.0");
+        try {
+            InputStream xml = new URL("https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/tools/devkit/mule-devkit-parent/maven-metadata.xml").openStream();
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(xml);
+            doc.getDocumentElement().normalize();
+            NodeList nList = doc.getElementsByTagName("version");
+            VersionUtils currentVersion;
+            String currentValue = "";
+            for (int i = 0; i < nList.getLength(); i++) {
+                Node tag = nList.item(i);
+                if (tag.getNodeType() == Node.ELEMENT_NODE) {
+                    Element version = (Element) tag;
+                    currentValue = version.getFirstChild().getTextContent();
+                }
+                /* It does not enter when the version has a - because those are test versions,
+               and Devkit version 4.x.x is not taken into account for being a migration tool */
+                if (currentValue.indexOf('-') < 0 && !currentValue.isEmpty() && currentValue.indexOf('4') != 0) {
+                    currentVersion = new VersionUtils(currentValue);
```

Comment:
These lines can be simplified to:

```
PomUtils.getCurrentDevKitVersion(currentValue).replaceIfGreaterThan(latestVersion);
```

(if you add the `new VersionUtils(currentValue)` to PomUtils, as mentioned in one of my other comments)